### PR TITLE
Jenkins is setting invalid versions

### PR DESCRIPTION
**Pull request by drewwells**
- clean up makefile.common targets
- move away from -atlas naming scheme
- cache docker image pushes

### CODE REVIEW

File: atlas/templates/Jenkinsfile.gotmpl [hunk 1]
```diff
@@ -54,7 +54,7 @@ pipeline {
             dir ("${WORKSPACE}/${DIRECTORY}") {
                 withDockerRegistry([credentialsId: "<insert-the-creds-id>", url: ""]) {
                     withAWS(region:'<insert-the-region-id>', credentials:'<insert-the-creds-id>') {
-                        sh "IMAGE_VERSION=\$(IMAGE_VERSION)-j$BUILD_NUMBER make push-chart"
```
Comment (kd7lxl) [inline atlas/templates/Jenkinsfile.gotmpl:57]: Yeah, this seems like a bug. `IMAGE_VERSION` needs to be consistent throughout `build` and `push-chart` stages. Good eye. 👍

File: atlas/templates/Makefile.common.gotmpl [hunk 2]
```diff
@@ -60,30 +60,28 @@ fmt-atlas:
...
 	@docker push $(SERVER_IMAGE):$(IMAGE_VERSION)
 endif
-
-.push-$(IMAGE_NAME)-$(IMAGE_VERSION):
-	$(MAKE) push-atlas
 	touch $@
 
 .PHONY: push
-push: .push-$(IMAGE_NAME)-$(IMAGE_VERSION)
+push: .docker-$(IMAGE_NAME)-$(IMAGE_VERSION) .push-$(IMAGE_NAME)-$(IMAGE_VERSION)
```
Comment (kd7lxl) [inline atlas/templates/Makefile.common.gotmpl:84]: I don't see how the extra level of abstraction solves anything, but it's not wrong.

File: atlas/templates/Makefile.common.gotmpl [hunk 3]
```diff
@@ -60,30 +60,28 @@ fmt-atlas:
...
 	@go test $(GO_TEST_FLAGS) $(GO_PACKAGES)
 
-docker-atlas:
-	@docker build -f $(SERVER_DOCKERFILE) -t $(SERVER_IMAGE):$(IMAGE_VERSION) .
+docker-atlas: .docker-$(IMAGE_NAME)-$(IMAGE_VERSION)
 	@docker image prune -f --filter label=stage=server-intermediate
 
 .docker-$(IMAGE_NAME)-$(IMAGE_VERSION):
-	$(MAKE) docker-atlas
+	@docker build -f $(SERVER_DOCKERFILE) -t $(SERVER_IMAGE):$(IMAGE_VERSION) .
```
Comment (burov) [inline atlas/templates/Makefile.common.gotmpl:67]: I found that usefull to build and push two images, image with strict tag and image with latest tag, what do you thing about adding this into atlas-cli ?

**Comment by Evgeniy-L** [comment]
closing as there are no actions for a long time

**Pull request closed without merging**